foxglove/cmd/config.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func newConfigCommand() *cobra.Command {
+	configCmd := &cobra.Command{
+		Use:   "config",
+		Short: "Manage CLI configuration",
+		Long: `Manage CLI configuration values.
+Available configuration keys:
+  - project-id: Default project ID for commands
+  - api-key: API key for authentication`,
+	}
+
+	configCmd.AddCommand(newConfigGetCommand())
+	configCmd.AddCommand(newConfigSetCommand())
+	configCmd.AddCommand(newConfigUnsetCommand())
+
+	return configCmd
+}
+
+func newConfigGetCommand() *cobra.Command {
+	getCmd := &cobra.Command{
+		Use:   "get [KEY]",
+		Short: "Get a configuration value",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			key := args[0]
+
+			if !isValidConfigKey(key) {
+				dief("Invalid configuration key '%s'. Valid keys are: project-id, api-key", key)
+			}
+
+			viperKey := mapConfigKeyToViperKey(key)
+			value := viper.GetString(viperKey)
+
+			if value == "" {
+				if viper.IsSet(viperKey) {
+					fmt.Println()
+				} else {
+					dief("Configuration key '%s' not found", key)
+				}
+			} else {
+				fmt.Println(value)
+			}
+		},
+	}
+	return getCmd
+}
+
+func newConfigSetCommand() *cobra.Command {
+	setCmd := &cobra.Command{
+		Use:   "set [KEY] [VALUE]",
+		Short: "Set a configuration value",
+		Args:  cobra.MaximumNArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) < 1 {
+				dief("Configuration key is required")
+			}
+
+			key := args[0]
+			if !isValidConfigKey(key) {
+				dief("Invalid configuration key '%s'. Valid keys are: project-id, api-key", key)
+			}
+
+			var value string
+			if len(args) == 2 {
+				value = args[1]
+			} else {
+				// If the value is not provided, prompt for an input
+				switch key {
+				case "project-id":
+					value = promptForInput(fmt.Sprintf("Enter project ID (will be written to %s):\n", viper.ConfigFileUsed()))
```

Comment:
The interactive mode surprised me; I expected this to error unless I included a value. I assume `configure-api-key` works this way to encourage private keys be kept out of history; do we want to adopt this for all config? Should api-key stay within "auth"?